code/modules/cm_aliens/structures/special/pylon_core.dm

```diff
@@ -151,9 +151,14 @@
 		if(linked_hive.hijack_burrowed_surge && (last_surge_time + surge_cooldown) < world.time)
 			last_surge_time = world.time
 			linked_hive.stored_larva++
+			linked_hive.hijack_burrowed_left--
 			announce_dchat("The hive has gained another burrowed larva! Use the Join As Xeno verb to take it.", src)
 			if(surge_cooldown > 30 SECONDS) //mostly for sanity purposes
 				surge_cooldown = surge_cooldown - surge_incremental_reduction //ramps up over time
+		if(linked_hive.hijack_burrowed_surge && linked_hive.hijack_burrowed_left < 1)
+			linked_hive.hijack_burrowed_surge = FALSE
+			xeno_message(SPAN_XENOANNOUNCE("Hive surge is fading away. You will no longer gain burrowed larva over time."), 3, linked_hive.hivenumber)
```

Comment:
The wording of this could be improved especially since its not currently called this in character at least as far as I can tell. E.g. "The hive's power wanes. You will no longer gain pooled larva over time."